mypy plugin support for dataclasses (#966)
samuelcolvin/pydantic: docs/mypy_plugin.md

#### Respect `Config.orm_mode`
* If `Config.orm_mode` is `False`, you'll get a mypy error if you try to call `.from_orm()`;
  cf. [ORM mode](usage/models.md#orm-mode-aka-arbitrary-class-instances)

#### Generate a signature for `dataclasses`

> * The [`dataclasses`](usage/dataclasses.md) is decorated class with `@pydantic.dataclasess.dataclass` which is inspected

Reviewer: ```suggestion
* classes decorated with [`@pydantic.dataclasess.dataclass`](usage/dataclasses.md) are type checked the same as standard python dataclasses
```

>  as python [dataclasses](https://docs.python.org/3/library/dataclasses.html) 

Reviewer: ```suggestion
```

> * The `@pydantic.dataclasess.dataclass` decorator accepts `Config` as a keyword argument which has the same meaning as [Config](usage/model_config.md).

Reviewer: ```suggestion
* The `@pydantic.dataclasess.dataclass` decorator accepts a `config` keyword argument which has the same meaning as [the `Config` sub-class](usage/model_config.md).
```
